src/Datastore/Operation.php

```diff
@@ -665,4 +665,30 @@ private function readOptions(array $options = [])
             'readOptions' => $readOptions
         ]);
     }
+
+    /**
+     * Sort entities into the order given in $keys.
+     *
+     * @param Entity[] $entities
+     * @param Key[] $keys
+     * @return Entity[]
+     */
+    private function sortEntities(array $entities, array $keys)
+    {
+        $res = [];
+        foreach ($keys as $key) {
+            $path = $key->path();
+
+            foreach ($entities as $arrayKey => $entity) {
+                $entityPath = $entity->key()->path();
+
+                if ($path === $entityPath) {
+                    $res[] = $entity;
+                    unset($entities[$arrayKey]);
```

Comment:
Just note that the return array above contains nulls for keys which don't exist. You may want to array_filter the result.